### PR TITLE
Implement additional usermigrations

### PR DESCRIPTION
**Pull request by lukasgraf**
This PR adds support for additional usermigrations using `ftw.usermigration`, and "officially" registers the already existing user migrations directly in `opengever.core`.

Additional user migrations implemented in this PR:

- **OGDS User references**
  Migrates references to user IDs in OGDS:
    - activity actors
    - notification userids
    - task principals
    -  task issuers
    - task responsibles

---

- **Plone Tasks**
  Migrates
    - Plone Task responsibles
    - Plone Task issuers
    - Plone Task responses (actors + changes)

---

- **Private Folders**
    - Folders themselves
    - Merges contents between old and new private folders if necessary

---

- **Checked out documents**
    - checked_out_by annotations + index
    - WebDAV Locks

---

- **Repository favorites**

---

In addition, it contains an optimization for the `DossierMigrator` so it doesn't unnecessarily reindex `responsible` for every dossier.

**TODO**:

- [x] Changelog
- [x] Better PR description
- [x] Testing on backport-branch

### CODE REVIEW

File: opengever/usermigration/ogds_references.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                moved = self._migrate_sql_column(
+                    Activity.__table__, 'actor_id',
+                    old_userid, new_userid)
+                self.activity_actors_moved.extend(moved)
+
+                # Migrate notification userids
+                moved = self._migrate_sql_column(
+                    Notification.__table__, 'userid',
+                    old_userid, new_userid)
+                self.notification_userids_moved.extend(moved)
```
Comment (phgross) [inline opengever/usermigration/ogds_references.py]: Watchers.actorid needs to migrated as well.

File: opengever/usermigration/ogds_references.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                    Task.__table__, 'issuer',
+                    old_userid, new_userid)
+                self.task_issuers_moved.extend(moved)
+
+                # Migrate task responsibles
+                moved = self._migrate_sql_column(
+                    Task.__table__, 'responsible',
+                    old_userid, new_userid)
+                self.task_responsibles_moved.extend(moved)
+
```
Comment (phgross) [inline opengever/usermigration/ogds_references.py]: What about OGDS Dictstorage entries?

File: opengever/usermigration/plone_tasks.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+from plone import api
+import logging
+
+
+logger = logging.getLogger('opengever.usermigration')
+
+FIELDS_TO_CHECK = ('responsible', 'issuer')
+
+
+class PloneTasksMigrator(object):
```
Comment (phgross) [inline opengever/usermigration/plone_tasks.py]: I would add a comment here, that all the automatic created local roles on tasks, task documents, and task containers are handled by another Migrator.

File: setup.py [hunk 4]
```diff
@@ -110,6 +110,7 @@
           'ftw.table>=1.18.0',
           'ftw.tika',
           'ftw.upgrade >= 1.18.0',
+          'ftw.usermigration',
```
Comment (phgross) [inline setup.py]: Please add also a version pinning to the `versions.cfg`

File: opengever/usermigration/ogds_references.py (hunk 2 above)
Comment (lukasgraf) [reply opengever/usermigration/ogds_references.py]: As discussed, already implemented in [`opengever/usermigration/dictstorage.py`](https://github.com/4teamwork/opengever.core/blob/master/opengever/usermigration/dictstorage.py)

**Comment by lukasgraf** [comment]
Thanks for the review! Updated - addressed all the feedback, added repo favorites migration that was missing, and fixed some issues with migration of task responses (see #3419).

**Pull request merged**